panther_diagnostics/include/panther_diagnostics/system_status.hpp

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PANTHER_DIAGNOSTICS_NODE_CPU_HPP
+#define PANTHER_DIAGNOSTICS_NODE_CPU_HPP
+
+#include <string>
+
+#include "diagnostic_updater/diagnostic_updater.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+#include "panther_msgs/msg/system_status.hpp"
+
+using namespace std::chrono_literals;
+
+namespace panther_diagnostics
+{
+
+class SystemStatus : public rclcpp::Node
+{
+public:
+  SystemStatus();
+
+  static constexpr const char * cpu_info_filename = "/proc/stat";
+  static constexpr const char * memory_info_filename = "/proc/meminfo";
+  static constexpr const char * temperature_info_filename = "/sys/class/thermal/thermal_zone0/temp";
```

Comment:
It generates warnings
![image](https://github.com/husarion/panther_ros/assets/109142865/531d3ed1-06f5-4162-9ec8-efef94620f81)

Reply:
Because you create a pointer to char. Try creating a char array like below:
```cpp
  static constexpr char kCPUInfoFilename[] = "/proc/stat";
  static constexpr char kMemoryInfoFilename[] = "/proc/meminfo";
  static constexpr char kTemperatureInfoFilename[] = "/sys/class/thermal/thermal_zone0/temp";
```